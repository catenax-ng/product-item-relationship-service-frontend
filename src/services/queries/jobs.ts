import { useMutation } from "@tanstack/react-query";
import { useQuery } from "../../lib/react-query";
import { IRSRequestBody } from "../../types/jobs";
import { getCurrentEnvironment } from "../../utils/sessionStorageHandling";
import { selectAPI } from "../api/jobs.api";

export const useFetchJobById = ({ id }: { id: string }, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, id], () => jobAPI.fetchJobById(id), {
    refetchInterval,
  });
};

export const useFetchJobs = (page: number, refetchInterval: false | number = false) => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useQuery(["jobs", serverEnv, page], () => jobAPI.fetchJobs(page), { refetchInterval });
};

export const useCancelJobs = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation(jobAPI.cancelJob);
};

export const useCreateJob = () => {
  const serverEnv = getCurrentEnvironment();
  const jobAPI = selectAPI(serverEnv);
  return useMutation({
    mutationFn: (data: IRSRequestBody) => {
      return jobAPI.createJob(data);
    },
  });
};
