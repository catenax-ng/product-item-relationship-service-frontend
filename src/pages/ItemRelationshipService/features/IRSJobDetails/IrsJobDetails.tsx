import { useFetchJobById } from "../../../../services/queries/jobs";
import { IRSJobTombstones } from "../IRSJobTombstones";
import { IrsJobDetailsDisplay } from "./IrsJobDetailsDisplay";

export const IrsJobDetails: React.FC<{ jobId: string }> = ({ jobId }) => {
  const { data: job, isError, isLoading } = useFetchJobById(jobId);
  if (isLoading) {
    //TODO: Handle Loading State
    console.warn("implement job loading");
    return null;
  }
  if (isError) {
    //TODO: handle Error state
    console.error("implement job error");
    return null;
  }

  return (
    <>
      <IrsJobDetailsDisplay job={job.job} />
      <IRSJobTombstones job={job} />
    </>
  );
};
