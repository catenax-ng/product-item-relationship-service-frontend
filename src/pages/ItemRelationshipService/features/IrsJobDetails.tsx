import { Box, Divider, Grid } from "@mui/material";
import dayjs from "dayjs";
import { useTranslation } from "react-i18next";
import { DetailGrid } from "../../../components/DetailGrid";
import { BeautifulJson } from "../../../lib/react-syntax-highlighter";
import { useFetchJobById } from "../../../services/queries/jobs";
import { Job } from "../../../types/jobs";

export const IrsJobDetails: React.FC<{ jobId: string }> = ({ jobId }) => {
  const { data: job, isError, isLoading } = useFetchJobById(jobId);
  if (isLoading) {
    //TODO: Handle Loading State
    console.warn("implement job loading");
    return null;
  }
  if (isError) {
    //TODO: handle Error state
    console.error("implement job error");
    return null;
  }

  return <IrsJobDetailsDisplay job={job.job} />;
};

export const IrsJobDetailsDisplay: React.FC<{ job?: Job }> = ({ job }) => {
  const { t } = useTranslation();

  return (
    <section
      style={{
        padding: "34px 20px",
      }}
    >
      <Box className="irs-job-details">
        <Box className="irs-job-details-header">
          <h5>{t("content.irs.jobDetails.title")}</h5>
        </Box>
        <Box className="irs-job-details-content">
          {job && (
            <Grid>
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid topic={"Job ID:"} content={job?.jobId} />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid topic={t("content.irs.jobDetails.globalAssetId")} content={job.globalAssetId} />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid topic={"Job State:"} content={job.jobState} />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid
                topic={t("content.irs.jobDetails.exception")}
                content={<BeautifulJson json={job.exception} />}
              />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid
                topic={t("content.irs.jobDetails.createdOn")}
                content={dayjs(job.createdOn).format("YYYY-MM-DD HH:mm:ss")}
              />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid
                topic={t("content.irs.jobDetails.startedOn")}
                content={dayjs(job.startedOn).format("YYYY-MM-DD HH:mm:ss")}
              />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid
                topic={t("content.irs.jobDetails.lastModifiedOn")}
                content={dayjs(job.lastModifiedOn).format("YYYY-MM-DD HH:mm:ss")}
              />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid
                topic={t("content.irs.jobDetails.jobCompleted")}
                content={dayjs(job?.jobCompleted).format("YYYY-MM-DD HH:mm:ss")}
              />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid topic={t("content.irs.jobDetails.owner")} content={job.owner} />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid topic={t("content.irs.jobDetails.summary")} content={<BeautifulJson json={job.summary} />} />
              <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
              <DetailGrid
                topic={t("content.irs.jobDetails.jobParameter")}
                content={<BeautifulJson json={job.jobParameter} />}
              />
            </Grid>
          )}

          {/* <Highlight className="irs-job-details-content-code">
              {JSON.stringify(job, null, 2)}
            </Highlight> */}
        </Box>
      </Box>
    </section>
  );
};
