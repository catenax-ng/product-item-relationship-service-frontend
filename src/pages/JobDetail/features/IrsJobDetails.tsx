import { Box, Divider, Grid } from "@mui/material";
import dayjs from "dayjs";
import { useTranslation } from "react-i18next";
import { DetailGrid } from "../../../components/DetailGrid";
import { BeautifulJson } from "../../../lib/react-syntax-highlighter";
import { Job } from "../../../types/jobs";

export const IrsJobDetails: React.FC<{ job: Job }> = ({ job }) => {
  const { t } = useTranslation();

  return (
    <section
      style={{
        padding: "34px 20px",
      }}
    >
      <Box className="irs-job-details">
        <Box className="irs-job-details-header">
          <h5>{t("content.irs.jobDetails.title")}</h5>
        </Box>
        <Box className="irs-job-details-content">
          <Grid>
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid topic={"Job ID:"} content={job?.jobId} />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid topic={t("content.irs.jobDetails.globalAssetId")} content={job.globalAssetId} />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid topic={"Job State:"} content={job.jobState} />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid
              topic={t("content.irs.jobDetails.exception")}
              content={<BeautifulJson json={job.exception} />}
            />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid
              topic={t("content.irs.jobDetails.createdOn")}
              content={dayjs(job.createdOn).format("YYYY-MM-DD HH:mm:ss")}
            />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid
              topic={t("content.irs.jobDetails.startedOn")}
              content={dayjs(job.startedOn).format("YYYY-MM-DD HH:mm:ss")}
            />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid
              topic={t("content.irs.jobDetails.lastModifiedOn")}
              content={dayjs(job.lastModifiedOn).format("YYYY-MM-DD HH:mm:ss")}
            />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid
              topic={t("content.irs.jobDetails.jobCompleted")}
              content={dayjs(job?.jobCompleted).format("YYYY-MM-DD HH:mm:ss")}
            />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid topic={t("content.irs.jobDetails.owner")} content={job.owner} />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid topic={t("content.irs.jobDetails.summary")} content={<BeautifulJson json={job.summary} />} />
            <Divider sx={{ mb: 2, mr: -2, ml: -2 }} />
            <DetailGrid
              topic={t("content.irs.jobDetails.jobParameter")}
              content={<BeautifulJson json={job.jobParameter} />}
            />
          </Grid>
        </Box>
      </Box>
    </section>
  );
};
