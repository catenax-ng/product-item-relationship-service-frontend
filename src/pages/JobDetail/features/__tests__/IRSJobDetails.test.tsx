import { JobsDemoDataSuccess } from "../../../../services/api/__test__/jobs.exampleData";
import { render } from "../../../../testing/test-utils";
import { IrsJobDetails } from "../IrsJobDetails";

it("renders the tombstone information", () => {
  const { container } = render(<IrsJobDetails job={JobsDemoDataSuccess.job} />);
  expect(container).toMatchSnapshot();
});
