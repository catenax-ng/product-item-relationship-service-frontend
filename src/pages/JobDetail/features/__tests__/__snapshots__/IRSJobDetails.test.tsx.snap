// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the tombstone information 1`] = `
<div>
  <section
    style="padding: 34px 20px;"
  >
    <div
      class="irs-job-details MuiBox-root css-0"
    >
      <div
        class="irs-job-details-header MuiBox-root css-0"
      >
        <h5>
          content.irs.jobDetails.title
        </h5>
      </div>
      <div
        class="irs-job-details-content MuiBox-root css-0"
      >
        <div
          class="MuiGrid-root css-vj1n65-MuiGrid-root"
        >
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              Job ID:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              e5347c88-a921-11ec-b909-0242ac120002
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.globalAssetId
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              urn:uuid:6c311d29-5753-46d4-b32c-19b918ea93b0
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              Job State:
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              COMPLETED
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.exception
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              <pre
                style="display: block; overflow-x: auto; padding: 0.5em; background: white; color: black;"
              >
                <code
                  style="white-space: pre;"
                >
                  <span>
                    {

                  </span>
                  <span>
                      "
                  </span>
                  <span
                    style="color: rgb(0, 0, 0);"
                  >
                    errorDetail
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ": "
                  </span>
                  <span>
                    Timeout while requesting Digital Registry
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ",

                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                      "
                  </span>
                  <span>
                    exception
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ": "
                  </span>
                  <span>
                    IrsTimeoutException
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ",

                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                      "
                  </span>
                  <span>
                    exceptionDate
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    ": "
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    2022
                  </span>
                  <span>
                    -
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    02
                  </span>
                  <span>
                    -
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    03
                  </span>
                  <span>
                    T
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    14
                  </span>
                  <span>
                    :
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    48
                  </span>
                  <span>
                    :
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    54
                  </span>
                  <span>
                    .
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    709
                  </span>
                  <span>
                    Z
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "

                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    }
                  </span>
                </code>
              </pre>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.createdOn
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.startedOn
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.lastModifiedOn
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.jobCompleted
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              2022-02-03 14:48:54
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.owner
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            />
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.summary
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              <pre
                style="display: block; overflow-x: auto; padding: 0.5em; background: white; color: black;"
              >
                <code
                  style="white-space: pre;"
                >
                  <span>
                    {

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "asyncFetchedItems"
                  </span>
                  <span>
                    : {

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "completed"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    0
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "failed"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    0
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "running"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    0
                  </span>
                  <span>
                    

                  </span>
                    }

                  }
                </code>
              </pre>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-1pdy5v-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-15v7om4-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
            >
              content.irs.jobDetails.jobParameter
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-vgqnfl-MuiGrid-root"
            >
              <pre
                style="display: block; overflow-x: auto; padding: 0.5em; background: white; color: black;"
              >
                <code
                  style="white-space: pre;"
                >
                  <span>
                    {

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "aspects"
                  </span>
                  <span>
                    : [

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "SerialPartTypization"
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                        
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "ContactInformation"
                  </span>
                  <span>
                    

                  </span>
                    ],

                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "bomLifecycle"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "asBuilt"
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "collectAspects"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    false
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "depth"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 102, 102);"
                  >
                    4
                  </span>
                  <span>
                    ,

                  </span>
                  <span>
                      
                  </span>
                  <span
                    style="color: rgb(102, 0, 102);"
                  >
                    "direction"
                  </span>
                  <span>
                    : 
                  </span>
                  <span
                    style="color: rgb(0, 136, 0);"
                  >
                    "downward"
                  </span>
                  <span>
                    

                  </span>
                  }
                </code>
              </pre>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
